syntax = "proto3";

message ToggleFollow {
  string Origin = 1;
  string Target = 2;
  bool Following = 3;
}

message ListUser {
  string Origin = 1;
}

message ReturnList {
  string CurrentUsers = 1;
  string Followers = 2;
}

message Post {
  string Origin = 1;
  string Post = 2;
  string Time = 3;
}

service TinySocialNetworkService {
  rpc Follow(ToggleFollow) returns (ToggleFollow) {}
  rpc Unfollow(ToggleFollow) returns (ToggleFollow) {}
  rpc List(ListUser) returns (ReturnList) {}
  rpc Timeline(stream Post) returns (stream Post) {}
}
